sold.cc

```diff
@@ -132,8 +134,10 @@ void Sold::BuildLoads() {
         }
```

Comment:
I added `tests/mprotect` to check the effect of this PR. As you can see below, it decreased the writable area. However, because of `load.emit.p_flags |= PF_W;`, there are two `RWX` areas. After I delete this line, I make asserts for this test case.

```
master                                                                                                                             size
7fffe7fbd000-7fffe7fbe000 rw-p 00000000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 1000
7fffe7fbe000-7ffff7fbd000 ---p 00001000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 
7ffff7fbd000-7ffff7fbe000 rw-p 00001000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 1000
7ffff7fbe000-7ffff7fbf000 rwxp 00002000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 1000
7ffff7fbf000-7ffff7fc3000 rw-p 00003000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 4000 <== 
7ffff7fc3000-7ffff7fc4000 rwxp 00007000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 1000
7ffff7fc4000-7ffff7fc8000 rw-p 00008000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 4000 <==


MprotectBuilder                                                                                                                    size
7fffe7fbc000-7fffe7fbd000 rw-p 00000000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 1000
7fffe7fbd000-7ffff7fbc000 ---p 00001000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 
7ffff7fbc000-7ffff7fbd000 rw-p 00001000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 1000
7ffff7fbd000-7ffff7fbe000 rwxp 00002000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 1000
7ffff7fbe000-7ffff7fbf000 rw-p 00003000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 1000 <==
7ffff7fbf000-7ffff7fc0000 r--p 00004000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 1000 <==
7ffff7fc0000-7ffff7fc2000 rw-p 00005000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 2000 <==
7ffff7fc2000-7ffff7fc3000 rwxp 00007000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 1000
7ffff7fc3000-7ffff7fc4000 rw-p 00008000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 1000 <==
7ffff7fc4000-7ffff7fc5000 r--p 00009000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 1000 <==
7ffff7fc5000-7ffff7fc7000 rw-p 0000a000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 2000 <==
7ffff7fc7000-7ffff7fc8000 r-xp 0000c000 fe:00 35812987                   /home/akira/sold/tests/mprotect/mprotect_check.so.soldout 1000
```